guidance/library/_json.py

```diff
@@ -99,6 +100,46 @@ class ObjectKeywords(str, Enum):
 
 WHITESPACE = {b" ", b"\t", b"\n", b"\r"}
 
+FORMAT_PATTERNS: dict[str, Optional[str]] = {
+    # https://json-schema.org/understanding-json-schema/reference/string#built-in-formats
+    # Dates and times
+    "date-time": r'\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(?:\.\d+)?(?:Z|[+-]\d{2}:\d{2})',
+    "time": r'\d{2}:\d{2}:\d{2}(?:\.\d+)?(?:Z|[+-]\d{2}:\d{2})?',
+    "date": r'\d{4}-\d{2}-\d{2}',
+    "duration": r'P(\d+Y)?(\d+M)?(\d+D)?(T(\d+H)?(\d+M)?(\d+S)?)?',
+    # Email addresses
```

Comment:
These are probably not comprehensive (per RFC 5322), but are a good starting point :-)